### ISSUE TITLE
Replace reexports with lifted versions of functions

### ISSUE THREAD

**Issue by snoyberg** [commercialhaskell/rio#55]
Functions specialized to `IO` should be generalized to either `MonadIO` or `MonadUnliftIO`. Applies at least to the modules `RIO.ByteString`, `RIO.Directory`, and `RIO.ByteString.Lazy`. Pointed out by @adituv in #53.

**Comment by adituv** [comment]
I have a few more functions here and there to add to the list: https://github.com/adituv/rio/compare/10054040c158c2b5cdfee8d47d241bafbd3ac2ad...adituv:55-lift-reexports

Happy to submit as a separate PR after #58 goes through, or equally happy for @roman to add them to that PR.

**Comment by snoyberg** [comment]
Thanks to you both! I'm going to reopen this, there are still a number of functions that need lifted versions.

**Comment by adituv** [comment]
Could you list some of the functions that need lifted versions?  That is, if you aren't planning to just fix them yourself.

**Comment by snoyberg** [comment]
I'd forgotten that I'd already added lifted versions for `RIO.ByteString`! Alright, scratch that, I'm not aware of others. Thanks!
